### ISSUE TITLE
[amp-story-player] Sometimes I see exception calling show(). 

### ISSUE THREAD

**Issue by pshihn** [ampproject/amphtml#31562]
## What's the issue?

Sometimes when I call `show()` on the player, it throws the following exception. It happens randomly and when I am calling `show()` on a url that's not next in the list.  This seems like some sort of race condition:

```
Uncaught TypeError: Cannot read property 'then' of undefined
    at ec (amp-story-player-impl.js:1316)
    at Eb.k.go (amp-story-player-impl.js:1307)
    at HTMLElement.onStoryNavigation (<anonymous>:3399:2887)
    at X.handleEvent (<anonymous>:29:12491)
    at HTMLElement.X.__boundHandleEvent (<anonymous>:29:11870)
    at Zb (amp-story-player-impl.js:841)
    at Eb.k.show (amp-story-player-impl.js:795)
````

## How do we reproduce the issue?

Create a player with lots of stories and randomly move between them.

## What browsers are affected?

All

## Which AMP version is affected?

Powered by AMP ⚡ HTML – Version 2011252111002

**Comment by pshihn** [comment]
Here's a reproducible case: https://jsfiddle.net/c056fu9y/8/
there are 23 stories in there. 
Click on `story 8` and then some other story and some other story. Toggling between the story after a couple of seconds on each story. You will see this exception and the story will not load.

**Comment by Enriqe** [comment]
Fixed by #32153
